src/compiler/checker.ts

```diff
@@ -12508,6 +12549,11 @@ namespace ts {
           * callable `then` signature.
           */
         function checkAsyncFunctionReturnType(node: FunctionLikeDeclaration): Type {
+            if (languageVersion >= ScriptTarget.ES6) {

```

Comment:
ES5 and earlier do not have a global Promise. We fall back to the existing behavior for ES5 as we plan to support async functions via tree transformation for target ES5 and users will need to be able to supply a Promise implementation.